### PR TITLE
Validation framework sync

### PR DESCRIPTION
**Pull request by balajipmlearn**
*Issue #, if available:*

*Description of changes:*
Aligns with EKS clusters created for the workshop

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: validation-frameworks/functional/README.md [hunk 1]
```diff
@@ -63,14 +63,23 @@ def test_check_logs():
 ### Running Functional Tests
 
 Follow these steps to run the functional tests:
-
-1. Navigate to the functional testing directory:
+1. Fetch kubeconfig for fleet-hub-cluster. Update the region in the below command accordingly
+```
+aws eks --region us-east-1 update-kubeconfig --name fleet-hub-cluster
```
Comment (csantanapr) [inline validation-frameworks/functional/README.md:74]: - Do not assume the region is `us-east-1` assume there is a variable `AWS_REGION` set.
- Validation should run on the spoke clusters not on the control plane (hub) cluster. Use cluster name `fleet-spoke-staging`
```suggestion
aws eks --region $AWS_REGION update-kubeconfig --name fleet-spoke-staging
```

File: validation-frameworks/functional/README.md [hunk 2]
```diff
@@ -63,14 +63,23 @@ def test_check_logs():
...
+2. Navigate to the functional testing directory:
 ```
 cd $VALIDATION_MODULE_HOME/functional/
 ```
-2. Run the pytest command:
+3. Install required packages:
+```
+python3 -m venv .venv
+source .venv/bin/activate
+pip install -r requirements.txt
```
Comment (csantanapr) [inline validation-frameworks/functional/README.md:84]: Mention that installing packages is only required if running outside workshop environment.

File: validation-frameworks/functional/test_endpoint.py [hunk 3]
```diff
@@ -7,19 +7,24 @@ def test_publish():
...
 def running_eks_cluster():
-    # Perform any necessary setup steps, such as deploying Prometheus in your EKS cluster
+    # Perform any necessary setup steps, such as deploying components in your EKS cluster
     pass
 
-#### WHEN RUNNING THIS PLEASE PORT FORWARD to 9090 ####
 @when("I check endpoints")
-def check_prometheus_endpoint():
+def check_eks_endpoint():
     eksclient = boto3.client('eks', region_name = 'us-east-1')
```
Comment (csantanapr) [inline validation-frameworks/functional/test_endpoint.py:15]: Do not hardcode region name, make sure an environment variable `AWS_REGION` is set

File: validation-frameworks/functional/test_endpoint.py [hunk 4]
```diff
@@ -7,19 +7,24 @@ def test_publish():
...
-    resp = eksclient.describe_cluster(name='blue')
+    resp = eksclient.describe_cluster(name='fleet-hub-cluster')
```
Comment (csantanapr) [inline validation-frameworks/functional/test_endpoint.py:16]: Do not hardcode the cluster name make sure an environment variable `AWS_EKS_CLUSTER_NAME` is set before starting tests

**Review by csantanapr — CHANGES_REQUESTED** [review]

File: validation-frameworks/functional/README.md (hunk 2 above)
Comment (balajipmlearn) [reply validation-frameworks/functional/README.md]: These are packages required for pytest BDD, which are required irrespective of the environment.

File: validation-frameworks/functional/test_endpoint.py (hunk 3 above)
Comment (balajipmlearn) [reply validation-frameworks/functional/test_endpoint.py]: Updated the code to fetch the region from AWS_REGION and cluster name from EKS_CLUSTER_NAME environment variables.

File: validation-frameworks/functional/test_endpoint.py (hunk 4 above)
Comment (balajipmlearn) [reply validation-frameworks/functional/test_endpoint.py]: Updated the code to fetch the region from AWS_REGION and cluster name from EKS_CLUSTER_NAME environment variables.

File: validation-frameworks/functional/README.md (hunk 2 above)
Comment (csantanapr) [reply validation-frameworks/functional/README.md]: I know @balajipmlearn they need to be installed but user during the workshop, in the IDE we provide can we preinstalled this packages for them?

Comment (csantanapr) [reply validation-frameworks/functional/README.md]: I can merge but we need to improve the experience of the workshop

**Review by csantanapr — APPROVED** [review]

**Comment by csantanapr** [comment]
there is something wrong with this PR, the area to merge is spinning
image

**Comment by csantanapr** [comment]
I was able to merge this in another PR  https://github.com/aws-samples/fleet-management-on-amazon-eks-workshop/pull/57

**Pull request closed without merging**